assembl/static/js/app/views/baseMessageList.js

```diff
@@ -0,0 +1,2366 @@
+   */
+  destroyAnnotator: function() {
+    if (!this.annotator) {
+      return;
+    }
+
+    this.trigger("annotator:destroy", this.annotator);
+
+    this.annotator.unsubscribe('annotationsLoaded');
+    this.annotator.unsubscribe('annotationCreated');
+    this.annotator.unsubscribe('annotationEditorShown');
+    this.annotator.unsubscribe('annotationViewerShown');
+
+    this.annotator.destroy();
+    this.annotator = null;
+  },
+
+  /**
+   * @event
+   * Shows all messages (clears all filters)
+   */
+  showAllMessages: function() {
+    //console.log("messageList:showAllMessages() called");
+    this.currentQuery.clearAllFilters();
+    this.render();
+  },
+
+  getTargetMessageViewStyleFromMessageListConfig: function(messageView) {
+    var defaultMessageStyle,
+        targetMessageViewStyle;
+
+    if (Ctx.getCurrentInterfaceType() === Ctx.InterfaceTypes.SIMPLE) {
+      defaultMessageStyle = Ctx.AVAILABLE_MESSAGE_VIEW_STYLES.FULL_BODY;
+    }
+    else {
+      defaultMessageStyle = this.defaultMessageStyle;
+    }
+
+    if (this.currentViewStyle === this.ViewStyles.NEW_MESSAGES) {
+      if (messageView.model.get('read') === true) {
+        targetMessageViewStyle = Ctx.AVAILABLE_MESSAGE_VIEW_STYLES.TITLE_ONLY;
+      }
+      else {
+        if (defaultMessageStyle !== Ctx.AVAILABLE_MESSAGE_VIEW_STYLES.TITLE_ONLY) {
+          targetMessageViewStyle = defaultMessageStyle;
+        }
+        else {
+          targetMessageViewStyle = Ctx.AVAILABLE_MESSAGE_VIEW_STYLES.FULL_BODY;
+        }
+      }
+    }
+    else {
+      targetMessageViewStyle = defaultMessageStyle;
+    }
+
+    return targetMessageViewStyle;
+  },
+
+  constrainViewStyle: function(viewStyle) {
+    if (!viewStyle) {
+      //If invalid, set global default
+      viewStyle = this.ViewStyles.RECENTLY_ACTIVE_THREADS;
+    }
+    return viewStyle;
+  },
+
+  /**
+   * @event
+   * Set the view to the selected viewStyle, if allowable by the current user
+   * Otherwise, sets the default style
+   * Does NOT re-render
+   *
+   */
+  setViewStyle: function(viewStyle, DEPRECATED_skip_storage) {
+      //console.log("setViewStyle called with: ", viewStyle, "interface type: ", Ctx.getCurrentInterfaceType(), "current user is unknown?:", Ctx.getCurrentUser().isUnknownUser());
+
+      viewStyle = this.constrainViewStyle(viewStyle);
+
+      if (this.isViewStyleThreadedType(viewStyle)) {
+        this.currentViewStyle = viewStyle;
+        this.currentQuery.setView(this.currentQuery.availableViews.THREADED);
+      }
+      else if (viewStyle === this.ViewStyles.REVERSE_CHRONOLOGICAL) {
+        this.currentViewStyle = viewStyle;
+        this.currentQuery.setView(this.currentQuery.availableViews.REVERSE_CHRONOLOGICAL);
+      } else if (viewStyle === this.ViewStyles.POPULARITY) {
+        this.currentViewStyle = viewStyle;
+        this.currentQuery.setView(this.currentQuery.availableViews.POPULARITY);
+      } else if (viewStyle === thi
```

Comment:
Ahhh....these parts of the messageList would have been great to Promisified. Oh well...